generative_data_prep/processors/article_tokenizer.py

```diff
@@ -334,10 +334,31 @@ def tokenize(self, completion: str, prompt: Optional[str] = None, category_id: O
             if self.prompt_postfix:
                 prompt = prompt + self.prompt_postfix
             prompt_token_ids = self.tokenizer.encode(prompt)
+            # If there is an EOS token at the end of the prompt then remove it
+            if (
+                hasattr(self.tokenizer, "eos_token_id")
+                and prompt_token_ids[-1] == self.tokenizer.eos_token_id
+                and len(prompt_token_ids) >= 1
+            ):
+                prompt_token_ids = prompt_token_ids[:-1]
             tokens += list(map(lambda x: Token(x, TokenTypeIds.PROMPT, category_id), prompt_token_ids))
 
         if completion:
             completion_token_ids = self.tokenizer.encode(completion)
+            # If there is a BOS token at the begining of the completion then remove it
+            if (
+                hasattr(self.tokenizer, "bos_token_id")
+                and completion_token_ids[0] == self.tokenizer.bos_token_id
+                and len(completion_token_ids) >= 1
```

Comment:
Similarly, let's also avoid removal if the prompt is empty (as is the case in pretraining).

Reply:
Good call! 🚀